how-to/misc/configuring-sriov.rst

```diff
@@ -0,0 +1,350 @@
+Configuring SR-IOV
+==================
+
+Overview
+--------
+
+The Single Root I/O Virtualization (SR-IOV) specification allows splitting
+a single physical port (called physical function or PF) into multiple virtual
+ports known as virtual functions or VFs.
+
+VFs as well as PFs can be assigned to Openstack instances, bypassing the
+host network stack and significantly improving performance while reducing
+host resource usage.
+
+Hardware offloading, also known as switchdev mode, is a mechanism that
+enables Open vSwitch (OVS) flows to be offloaded directly to the Virtual
+Function (VF). This approach combines the performance advantages of SR-IOV
+with the flexibility of OVS, allowing VFs to be added to the OVS bridge. As
+a result, it supports the full OVN stack and facilitates overlay tenant
+networks that rely on tunneling protocols such as VXLAN or Geneve.
+
+.. _sriov_prerequisites:
+
+Prerequisites
+-------------
+
+1. Make sure that the network adapters support SR-IOV and optionally the
+hardware offloading feature.
+
+2. Enable the SR-IOV and VT-d settings in BIOS.
+
+3. Add the following kernel parameters, allowing VFs to be exposed to virtual
+machines.
+
+::
+
+    intel_iommu=on iommu=pt pci=realloc pci=assign-busses
+
+4. Optional: enable switchdev mode.
+
+In this example, we are configuring a Mellanox ConnectX-6 device. Please
+check your vendor documentation for other network adapter models.
+
+::
+
+    ifname=enp3s0f0np0
+    pciaddr=0000:02:00.0
+    sudo devlink dev eswitch set pci/${pciaddr} mode switchdev
+    sudo ethtool -K $ifname hw-tc-offload on
+
+Verify that hardware offloading is enabled:
+
+::
+
+    sudo devlink dev eswitch show pci/${pciaddr}
+    # output #
+    pci/0000:03:00.0: mode switchdev inline-mode none encap-mode basic
+
+::
+
+    sudo ethtool -k $ifname | grep hw-tc-offload
```

Comment:
For the record, there's a Netplan feature request for setting `hw-tc-offload`:

https://bugs.launchpad.net/netplan/+bug/2070314